packages/wallets/src/components/WalletTransactions/WalletTransactions.tsx

```diff
@@ -1,32 +1,60 @@
-import React, { ComponentProps, useState } from 'react';
-import { WalletTransactionsCrypto } from '../WalletTransactionsCrypto';
+import React, { ComponentProps, useEffect, useState } from 'react';
+import { useActiveWalletAccount } from '@deriv/api';
+import { WalletTransactionsCompleted } from '../WalletTransactionsCompleted';
 import { WalletTransactionsFilter } from '../WalletTransactionsFilter';
+import { WalletTransactionsPending } from '../WalletTransactionsPending';
 import './WalletTransactions.scss';
 
+type TWalletTransactionsPendingFilter = ComponentProps<typeof WalletTransactionsPending>['filter'];
+type TWalletTransactionCompletedFilter = ComponentProps<typeof WalletTransactionsCompleted>['filter'];
+
 const WalletTransactions = () => {
-    const [filterValue, setFilterValue] =
-        useState<ComponentProps<typeof WalletTransactionsCrypto>['filter']>(undefined);
-    const [isPendingActive, setIsPendingActive] = useState(true);
+    const { data } = useActiveWalletAccount();
+    const [isPendingActive, setIsPendingActive] = useState(false);
+    const [filterValue, setFilterValue] = useState<
+        TWalletTransactionCompletedFilter | TWalletTransactionsPendingFilter
+    >();
+
+    useEffect(() => {
+        if (isPendingActive && filterValue === 'transfer') {
```

Comment:
I see this useEffect being added to mitigate the differences in cashier_payments and statement apis. Probably best to put them to respective WalletTransactionsCompleted and WalletTransactionsPending component, as those components are responsible for handling the data fetching and they should understand the nitty-gritty of API differences.
WalletTransactions component is not fetching or handling the data, thus, shouldn't be responsible fore handling details of API differences.